plugins/toolbox/toolbox.plugin.zsh

```diff
@@ -1,5 +1,110 @@
+if [[ $? -eq 0 ]] && ! type toolbox &>/dev/null; then
+  print "[oh-my-zsh] toolbox plugin: shell function 'toolbox' not defined.\n" \
+    "Please check toolbox" >&2
+  return
+fi
+
+function is_in_toolbox() {
+  [[ -f /run/.containerenv && -f /run/.toolboxenv ]]
+}
+
 function toolbox_prompt_info() {
-  [[ -f /run/.toolboxenv ]] && echo "⬢"
+  if is_in_toolbox; then
+    echo "⬢ "
+  fi
+}
+
+function toolbox_name {
+  # Get absolute path, resolving symlinks
+  local PROJECT_ROOT="${PWD:A}"
+  while [[ "$PROJECT_ROOT" != "/" && ! -e "$PROJECT_ROOT/.toolbox" &&
+    ! -d "$PROJECT_ROOT/.git" ]]; do
+    PROJECT_ROOT="${PROJECT_ROOT:h}"
+  done
+
+  # Check for toolbox name override
+  if [[ -f "$PROJECT_ROOT/.toolbox" ]]; then
+    echo "$(<"$PROJECT_ROOT/.toolbox")"
+  elif [[ -d "$PROJECT_ROOT/.git" && -n "$TOOLBOX_DEFAULT_CONTAINER" ]]; then
+    echo "$TOOLBOX_DEFAULT_CONTAINER"
+  fi
 }
 
-alias tb="toolbox enter"
+
+# Automatically enter Git projects in the default toolbox container. Toolbox
+# container can be overridden by placing a .toolbox file in the project root
+# with a container name in it.
+#
+function toolbox_cwd {
+  if [[ -z "$TOOLBOX_CWD" ]]; then
+    local TOOLBOX_CWD=1
+    local TOOLBOX_NAME=$(toolbox_name)
+
+    if [[ -n "$TOOLBOX_NAME" ]]; then
+      if ! is_in_toolbox; then
+        if ! $(podman container exists "$TOOLBOX_NAME"); then
+          tbc "$TOOLBOX_NAME"
+        fi
+        toolbox --container "$TOOLBOX_NAME" run sudo hostname "$TOOLBOX_NAME"
+        toolbox enter "$TOOLBOX_NAME"
+      fi
+    elif [[ "$(hostname)" != "toolbox" && $DISABLE_TOOLBOX_EXIT -ne 1 ]]; then
+      if is_in_toolbox; then
+        exit
+      fi
+    fi
+  fi
+}
```

Comment:
Could you describe what this does in detail, and where it is run? I doubt whether the exit will be reached at any point in time, and also I can't figure out when it is checked that we have cd-ed outside of the project root.